packages/nuxt/src/core/plugins/detect-component-usage.ts

```diff
@@ -0,0 +1,39 @@
+import { createUnplugin } from 'unplugin'
+import { isAbsolute, join, relative } from 'pathe'
+
+interface DetectComponentUsageOptions {
+  rootDir: string
+  exclude?: Array<RegExp | string>
+  detectedComponents: Set<string>
+}
+
+export const DetectComponentUsagePlugin = (options: DetectComponentUsageOptions) => createUnplugin(() => {
+  const importersToExclude = options?.exclude || []
+
+  const detectComponentUsagePatterns: Array<[importPattern: string | RegExp, name: string]> = [
+    ['../packages/nuxt/src/pages/runtime/page', 'NuxtPage'],
+    ['../packages/nuxt/src/app/components/nuxt-layout', 'NuxtLayout']
```

Comment:
these paths will work in the framework repo + tests but won't work in a user project as they will be `dist` paths instead. We expose the `distDir` in `src/dirs` however so you could use that.